text/0000-ques-in-main.md

```diff
@@ -224,111 +428,39 @@ for some value of `ErrorT` TBD.  It doesn't need to parse the body of
 the test to know whether it should do this; it can just do it
 unconditionally.
 
-## Standard impls of Termination
-[standard-impls-of-termination]: #standard-impls-of-termination
+## New constructors for `ExitStatus`
+[exit-status]: #exit-status
 
-At least the following implementations of Termination are available in
-libstd.  I use the ISO C constants `EXIT_SUCCESS` and `EXIT_FAILURE`
-for exposition.  [See below][unix-specific-refinements] for more
-discussion of these constants and the `TermStatus` type.
+As mentioned above, we propose to reuse `process::ExitStatus` as a way
+for a program to generate both success and failure conditions
+_without_ any errors printed, by returning it from `main`.  To make
+this more convenient, we also propose to add the following new
+constructors to `ExitStatus`:
 
 ``` rust
-impl Termination for ! {
-    fn write_diagnostic(&self, progname: &str, stream: &mut Write) -> ()
-    { unreachable!(); }
-    fn exit_status(&self) -> i32
-    { unreachable!(); }
-}
-
-impl Termination for () {
-    fn write_diagnostic(&self, progname: &str, stream: &mut Write) -> ()
-    { }
-    fn exit_status(&self) -> i32
-    { EXIT_SUCCESS }
-}
-
-impl Termination for TermStatus {
-    fn write_diagnostic(&self, progname: &str, stream: &mut Write) -> ()
-    { }
-    fn exit_status(&self) -> i32
-    { self.0 }
-}
-
-impl<E: Error> Termination for E {
-    fn write_diagnostic(&self, progname: &str, stream: &mut Write) -> ()
-    {
-        // unspecified, but not entirely unlike this:
-        if let Some(ref cause) = self.cause() {
-            cause.write_diagnostic(progname, stream);
-        }
-        writeln!(stream, "{}: {}\n", progname, self.description());
-    }
-
-    fn exit_status(&self) -> i32
-    { EXIT_FAILURE }
-}
-
-impl<T: Termination, E: Termination> Termination for Result<T, E> {
-    fn write_diagnostic(&self, progname: &str, stream: &mut Write) {
-        match *self {
-            Ok(ref ok) { ok.write_diagnostic(progname, stream); },
-            Err(ref err) { err.write_diagnostic(progname, stream); }
-        }
-    }
-    fn exit_status(&self) -> i32 {
-        match *self {
-            Ok(ref ok) => ok.exit_status(),
-            Err(ref err) => match err.exit_status() {
-                0 | EXIT_SUCCESS => EXIT_FAILURE,
-                e => e
-            }
-        }
-    }
+impl ExitStatus {
+    /// Return an ExitStatus value representing success.
+    /// (ExitStatus::ok()).success() is guaranteed to be true, and
+    /// (ExitStatus::ok()).code() is guaranteed to be Some(0).
+    pub fn ok() -> Self;
```

Comment:
I'm all for adding these constructors, but as a bikesheddy nit, `ok()` reads oddly here, because it so commonly refers to `Option::ok()`.  I'd have suggested `success()` if that wasn't already a non-constructor function.  `pass()` perhaps, by contrast with `failure()`? (Better suggestion welcome, please.)